File: codes/week18/진현/374.cpp
#include <bits/stdc++.h>

using namespace std;

/**
 * Forward declaration of guess API.
 * @param  num   your guess
 * @return 	     -1 if num is higher than the picked number
 *			      1 if num is lower than the picked number
 *               otherwise return 0
 * int guess(int num);
 */

int guess(int num);

class Solution {
public:
    int guessNumber(int n) {
        int mid, l = 1, r = n;

        while (1) {
            mid = l + (r - l) / 2;

            int ret = guess(mid);
            if (ret == -1)
                r = mid - 1;
            else if (ret == 1)
                l = mid + 1;
            else
                break;
        }
        return mid;
    }
};

File: codes/week18/지원/374.py
# The guess API is already defined for you.
# @param num, your guess
# @return -1 if num is higher than the picked number
#          1 if num is lower than the picked number
#          otherwise return 0
# def guess(num):

class Solution(object):
    def guessNumber(self, n):
        """
        :type n: int
        :rtype: int
        """
        s = 1
        e = n

        while True:
            m = (s + e) // 2

            if guess(m) == 0:
                return m
            elif guess(m) == -1:
                e = m - 1
            else:
                s = m + 1
        